dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -175,6 +178,73 @@ public WorkflowDAO(SessionFactory factory) {
         return filteredWorkflows.size() == 1 ? Optional.of(filteredWorkflows.get(0)) : Optional.empty();
     }
 
+    @SuppressWarnings({"checkstyle:ParameterNumber"})
+    public List<Workflow> filterTrsToolsGet(String descriptorType, String registry, String organization, String name, String toolname,
+            String description, String author, Boolean checker) {
+
+        SourceControlConverter converter = new SourceControlConverter();
+        CriteriaBuilder cb = currentSession().getCriteriaBuilder();
+        CriteriaQuery<Workflow> q = cb.createQuery(Workflow.class);
+        Root<Workflow> entryRoot = q.from(Workflow.class);
+
+        List<Predicate> predicates = new ArrayList<>();
+        predicates.add(cb.isTrue(entryRoot.get("isPublished")));
+        Predicate predicate = cb.isTrue(entryRoot.get("isPublished"));
+
+        predicate = andLike(cb, predicate, entryRoot.get("organization"), Optional.ofNullable(organization));
+        predicate = andLike(cb, predicate, entryRoot.get("repository"), Optional.ofNullable(name));
+        predicate = andLike(cb, predicate, entryRoot.get("workflowName"), Optional.ofNullable(toolname));
+        predicate = andLike(cb, predicate, entryRoot.get("description"), Optional.ofNullable(description));
+        predicate = andLike(cb, predicate, entryRoot.get("author"), Optional.ofNullable(author));
+        predicate = andEqualDescriptorType(cb, predicate, entryRoot.get("descriptorType"), Optional.ofNullable(descriptorType));
+
+        if (registry != null) {
+            predicate = cb.and(predicate, cb.equal(entryRoot.get("sourceControl"), converter.convertToEntityAttribute(registry)));
+        }
+
+        if (checker != null) {
+            predicate = cb.and(predicate, cb.isTrue(entryRoot.get("isChecker")));
+        }
+
+        q.where(predicate);
+        TypedQuery<Workflow> query = currentSession().createQuery(q);
+
+        List<Workflow> workflows = query.getResultList();
+        return workflows;
+    }
+
+    private Predicate andLike(CriteriaBuilder cb, Predicate existingPredicate, Path<String> column, Optional<String> value) {
+        return value.map(val -> cb.and(existingPredicate, cb.like(column, wildcardLike(val))))
+                .orElse(existingPredicate);
+    }
+
+    private String wildcardLike(String value) {
+        return '%' + value + '%';
+    }
+
+    private Predicate andEqualDescriptorType(CriteriaBuilder cb, Predicate existingPredicate, Path<String> column, Optional<String> value) {
+        DescriptorLanguage descriptorLanguage;
+        if (value.isPresent()) {
+            String descriptorValue = value.get();
```

Comment:
Pretty sure you can just invoke `DescriptorLanguage.convertShortStringToEnum` (although you'll need to catch the exception).